pacsman/utils.py

```diff
@@ -62,7 +62,9 @@ def set_undefined_tags_to_blank(dataset, additional_tags):
 
 def copy_dicom_attributes(destination_dataset, source_dataset, additional_tags):
     for tag in additional_tags or []:
-        setattr(destination_dataset, tag, dataset_attribute_fetcher(source_dataset, tag))
+        value = dataset_attribute_fetcher(source_dataset, tag)
+        if value is not None:
```

Comment:
If the problem is only copying a `None` value to a dataset where the attribute is already present, shouldn't this be `if value is not None and getattr(destination_dataset, tag, None) is not None`?
As it is, `None` values won't ever be copied, even if the attribute is not present.

Reply:
Good question. I will probe with a unit test or two.

Reply:
I think the code is correct as is.

The problem being solved is that this code will fail with an exception: `setattr(destination_dataset, 'PatientName', None)`
So the test `if value is not None` protects against that failing case.

For these DICOM data sets, the value 'None' has the semantic meaning of being missing from the data. The code then says: do not erase from the destination an attribute is that is missing in the the source.

The method `copy_dicom_attributes` has two uses.
The first use case is to create a dataset which is a pared down version of another data set. This starts then with a source data set, and then copies into an empty dataset just those values which have been requested.
The second use case is to merge two datasets from two different levels, say study and series or series and instance. The goal is to have a lower level data set that retains some values found only in the upper level. An example of this is adding to a instance (slice) dataset the number of slices attribute found in the series level but not the instance.

In the first use case an attribute missing from the source will produce a value of `None` and then nothing is changed in the destination. This is correct behavior.
In the first use case, if the attribute exists in the source, then it is copied into the destination. This is also correct behavior.

In the second use case, if the attribute does not exist in the source, then it is left alone in the destination. This is correct merge behavior.
In the second use case, if the attribute exists in the source, but not in the destination, it is written to the destination. This is correct merge behavior. 
If the attribute exists in both the source and the destination, then the source wins. In the typical case both source and destination have the same value any way, so this is correct merge. The atypical case, where source and destination have different values, probably means there is a problem somewhere else. But at least the outcome is what one would expect from a "copy".

Reply:
I think this is OK. The problem I'm thinking of is that `None` has the meaning of being missing from the data, and so does the attribute not being present at all. But users of the datasets already need to be checking for this.